### PR TITLE
mgr/dashboard: use grafana server instead of grafana-server in grafana 10.4.0

### PR DESCRIPTION
**Pull request by aaSharma14**
The grafana-server command is deprecated in grafana v10.4.0. It is advised to use grafan server in place of it.

Fixes: https://tracker.ceph.com/issues/65479

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by cloudbehl** [comment]
tracker?

**Comment by aaSharma14** [comment]
jenkins test windows

**Review by afreen23 — COMMENTED** [review]
I am having one question , I can see that we are using v9 for grafana and the change we are proposing is for 10.4.0. I hope I am not missing something.

https://github.com/ceph/ceph/blob/e4c1c9a8c2b3d9f7910b2efcdee999b35be2b2ae/src/cephadm/cephadmlib/constants.py#L12

cc @aaSharma14

**Comment by aaSharma14** [comment]
> I am having one question , I can see that we are using v9 for grafana and the change we are proposing is for 10.4.0. I hope I am not missing something.
> 
> https://github.com/ceph/ceph/blob/e4c1c9a8c2b3d9f7910b2efcdee999b35be2b2ae/src/cephadm/cephadmlib/constants.py#L12
> 
> cc @aaSharma14

@afreen23 , Grafana version is being updated here in this PR - https://github.com/ceph/ceph/pull/56331

File: src/cephadm/cephadm.py [hunk 1]
```diff
@@ -3580,7 +3580,7 @@ def list_daemons(
                                 elif daemon_type == 'grafana':
                                     out, err, code = call(ctx,
                                                           [container_path, 'exec', container_id,
-                                                           'grafana-server', '-v'],
+                                                           'grafana server', '-v'],
```
[the remark this answers is not in the document]
Comment (aaSharma14) [reply src/cephadm/cephadm.py]: Thanks @adk3798 , updated the PR

**Review by ktdreyer — APPROVED** [review]

**Comment by cloudbehl** [comment]
> > I am having one question , I can see that we are using v9 for grafana and the change we are proposing is for 10.4.0. I hope I am not missing something.
> > https://github.com/ceph/ceph/blob/e4c1c9a8c2b3d9f7910b2efcdee999b35be2b2ae/src/cephadm/cephadmlib/constants.py#L12
> > 
> > cc @aaSharma14
> 
> @afreen23 , Grafana version is being updated here in this PR - #56331

Does this mean that we need to merge #56331 before we merge this? Does `grafana server -v` works with 9.4.x?

**Comment by nizamial09** [comment]
I merged the grafana version bump PR. So probably we can take this in after testing it.

**Comment by aaSharma14** [comment]
jenkins test make check

**Comment by aaSharma14** [comment]
jenkins test make check

**Comment by nizamial09** [comment]
@adk3798 probably we can look into merging this.

**Review by adk3798 — APPROVED** [review]
code LGTM

**Pull request merged**